packages/opentelemetry-sdk-trace-base/src/TimedEvent.ts

```diff
@@ -26,4 +26,6 @@ export interface TimedEvent {
   name: string;
   /** The attributes of the event. */
   attributes?: SpanAttributes;
+  /** Count of attributes of the event that were dropped due to collection limits */
+  droppedAttributesCount?: number;
```

Comment:
Is there any situation where this is `undefined`? To my understanding it's always initialized in `addEvent()`.

Or is it needed to add it optional because of backward compatibility?

Reply:
As I mentioned in another comment, I've made it optional because `attributes` is optional. So having the count required would not make sense.

Reply:
The optionality of both of these properties comes from the underlying proto definition